cli/azd/pkg/auth/manager.go

```diff
@@ -590,7 +611,68 @@ func (m *Manager) saveUserProperties(user *userProperties) error {
 		return fmt.Errorf("setting account id in config: %w", err)
 	}
 
-	return m.configManager.Save(cfg)
+	return m.saveAuthConfig(cfg)
+}
+
+// readAuthConfig loads the configuration from [cAuthConfigFileName] and returns a parsed version of it. If the config
+// file does not exist, an empty [config.Config] is returned, with no error.
+func (m *Manager) readAuthConfig() (config.Config, error) {
+	cfgPath, err := config.GetUserConfigDir()
+	if err != nil {
+		return nil, fmt.Errorf("getting user config dir: %w", err)
+	}
+
+	authCfgFile := filepath.Join(cfgPath, cAuthConfigFileName)
+
+	cfg, err := m.configManager.Load(authCfgFile)
+	if err == nil {
+		return cfg, nil
+	} else if !errors.Is(err, os.ErrNotExist) {
+		return nil, fmt.Errorf("reading auth config: %w", err)
+	}
+
+	// We used to store auth related configuration in the user configuration file directly. If above file did not exist,
+	// see if there is the data in the old location, and if so migrate it to the new location. This upgrades the old
+	// format to the new format.
+	cfg, err = m.userConfigManager.Load()
+	if err != nil {
+		return nil, fmt.Errorf("reading user config: %w", err)
+	}
+
+	curUserData, has := cfg.Get(cCurrentUserKey)
+	if !has {
+		return config.NewConfig(map[string]any{}), nil
+	}
+
+	if err := cfg.Unset(cCurrentUserKey); err != nil {
```

Comment:
Can we unset this after we've migrated the old value?

Reply:
> Nvm, I think unset changes the in-memory object but doesn't persist. Feel free to close this if that's your understanding as well.

Yes, that's my understanding as well.

>  It's weird that unset returns error

Yeah, it is.  I guess in some ways you could imagine trying to unset an invalid key like `foo..bar` or something and that leading to an error.